README.md

```diff
@@ -0,0 +1,306 @@
+---
+title: Nested dependencies in frontend
+published: false
+description: What are nested dependencies, why they exist, how they can harm the frontend and what you can do to solve it.
+tags: javascript, import-maps, node, rollup
+---
+
+So you got this awesome idea and you want to now actually do it.
+I am pretty sure you do not want to start from scratch so let's use existing open source packages.
+
+If you want to play along - all the code is on [github](https://github.com/daKmoR/nested-dependecies-in-frontend).
+
+For our example case, we wanna use lit-element and lit-html.
+
+```bash
+mkdir nested-dependecies-in-frontend
+cd nested-dependecies-in-frontend
+npm install lit-element lit-html@1.0.0 --save-exact
+```
+
+> we are using pinned versions on purpose here
+
+and then we just load both packages in our `main.js`.
+```js
+import { LitElement } from "lit-element";
+import { html } from "lit-html";
+
+console.log(LitElement);
+console.log(html);
+```
+
+In order to find out how big our app will be, we would like to create a rollup bundle.
+
+```bash
+npm install -D rollup
+```
+and create a `rollup.config.js`
+```js
+export default {
+  input: "main.js",
+  output: {
+    file: "bundle.js",
+    format: "iife"
+  },
+};
+```
+and add a `"build": "rollup -c rollup.config.js && du -h bundle.js"` to our package.json so we build the file and output it's file size.
+Let's run it :)
+
+oh it doesn't work :sob:
+
+```
+(!) Unresolved dependencies
+https://rollupjs.org/guide/en#warning-treating-module-as-external-dependency
+lit-element (imported by main.js)
+lit-html (imported by main.js)
+```
+
+ok, I heard that before... we need to add some plugins to get it to understand the way node resolution works.
```

Comment:
```suggestion

OK, I've heard this one before... We need to add some plugins so that Rollup will understand the way node resolution (i.e. bare module specifiers like `import { html } from 'lit-html'`) works.
```